src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -144,7 +145,8 @@ public class ImageSharpMiddleware
             ICacheHash cacheHash,
             CommandParser commandParser,
             FormatUtilities formatUtilities,
-            AsyncKeyReaderWriterLock<string> asyncKeyLock)
+            AsyncKeyReaderWriterLock<string> asyncKeyLock,
+            ImageSharpRequestAuthorizationUtilities requestAuthorizationUtilities)
```

Comment:
Adding this parameter is a (binary) breaking change which can be avoided by re-adding the previous constructor as obsolete and manually constructing the `ImageSharpRequestAuthorizationUtilities`:

```c#
[Obsolete]
public ImageSharpMiddleware(
    RequestDelegate next,
    IOptions<ImageSharpMiddlewareOptions> options,
    ILoggerFactory loggerFactory,
    IRequestParser requestParser,
    IEnumerable<IImageProvider> resolvers,
    IEnumerable<IImageWebProcessor> processors,
    IImageCache cache,
    ICacheKey cacheKey,
    ICacheHash cacheHash,
    CommandParser commandParser,
    FormatUtilities formatUtilities,
    AsyncKeyReaderWriterLock<string> asyncKeyLock)
    : this(
        next,
        options,
        loggerFactory,
        requestParser,
        resolvers,
        processors,
        cache,
        cacheKey,
        cacheHash,
        commandParser,
        formatUtilities,
        asyncKeyLock,
        new ImageSharpRequestAuthorizationUtilities(options, requestParser, processors, commandParser, null))
{
}
```

This does require `ImageSharpRequestAuthorizationUtilities` to allow a null value for `IServiceProvider`, but that shouldn't be an issue, as it isn't required/used by the methods this middleware calls into anyway.

Reply:
While looking into this a bit more, I noticed `ImageSharpRequestAuthorizationUtilities` is currently only really required to strip unknown commands, as `ComputeHMACAsync(ImageCommandContext)` just delegates to `this.options.OnComputeHMACAsync(ImageCommandContext, byte[])`. I also think that stripping unknown commands isn't a request authorization concern, so it's probably best to remove this dependency on `ImageSharpRequestAuthorizationUtilities` altogether...